### PR TITLE
fix: Moodle duplicate content metadata records detected

### PR DESCRIPTION
**Pull request by johnnagro**
When circumstances result in the Moodle client trying to create a duplicate course we should handle that error as a success (when we can).

- [ENT-5098](https://openedx.atlassian.net/browse/ENT-5098)

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: integrated_channels/moodle/client.py [hunk 1]
```diff
@@ -122,23 +135,26 @@ def _post(self, additional_params, method_url=None):
             'moodlewsrestformat': 'json',
         }
         params.update(additional_params)
-        if method_url:
```
Comment (johnnagro) [inline integrated_channels/moodle/client.py:125]: the `method_url` stuff wasn't in use

File: integrated_channels/moodle/client.py [hunk 2]
```diff
@@ -20,6 +20,18 @@
 MOODLE_FINAL_GRADE_ASSIGNMENT_NAME = '(edX integration) Final Grade'
 
 
+class MoodleClientError(ClientError):
```
Comment (alex-sheehan-edx) [inline integrated_channels/moodle/client.py:23]: 🎉 love the expansion on ClientErrors

File: integrated_channels/moodle/client.py [hunk 3]
```diff
@@ -331,10 +346,23 @@ def create_content_metadata(self, serialized_data):
           courses[1][fullname]: 'value',
           [...]
         }
+        when sending 1 course and its a dupe, treat as success.
+        when senting N courses and a dupe exists, throw exception since
```
Comment (alex-sheehan-edx) [inline integrated_channels/moodle/client.py:350]: NIT: `senting` -> `sending`

File: tests/test_integrated_channels/test_moodle/test_client.py [hunk 4]
```diff
@@ -80,6 +101,47 @@ def test_get_course_id(self):
         )
         assert client.get_course_id('course:test_course') == 2
 
+    def test_successful_create_content_metadata(self):
+        """
+        Test core logic of create_content_metadata to ensure
+        query string we send to Moodle is formatted correctly.
+        """
+        expected_data = SERIALIZED_DATA.copy()
+        # expected_data['courses[0][id]'] = self.moodle_course_id
```
Comment (alex-sheehan-edx) [inline tests/test_integrated_channels/test_moodle/test_client.py:110]: cruft?

File: tests/test_integrated_channels/test_moodle/test_client.py [hunk 5]
```diff
@@ -80,6 +101,47 @@ def test_get_course_id(self):
...
+        expected_data['wsfunction'] = 'core_course_create_courses'
+
+        client = MoodleAPIClient(self.enterprise_config)
+        client._post = unittest.mock.MagicMock(name='_post', return_value=SUCCESSFUL_RESPONSE)  # pylint: disable=protected-access
```
Comment (alex-sheehan-edx) [inline tests/test_integrated_channels/test_moodle/test_client.py:114]: just a thought- is there an easy way we can disable this pylint warning for all test files?

File: tests/test_integrated_channels/test_moodle/test_client.py [hunk 6]
```diff
@@ -23,13 +23,34 @@
...
+    'courses[0][startdate]': '2030-01-01T00:00:00Z',
+    'courses[0][enddate]': '2030-03-01T00:00:00Z',
+    'courses[1][summary]': 'edX Demonstration Course 2',
+    'courses[1][shortname]': 'edX Demonstration Course 2 (edX+DemoX)',
+    'courses[1][idnumber]': 'edX+DemoX2',
+    'courses[1][startdate]': '2030-01-01T00:00:00Z',
+    'courses[1][enddate]': '2030-03-01T00:00:00Z',
+}
+
 # MOODLE_COURSE_ID = 100
```
Comment (alex-sheehan-edx) [inline tests/test_integrated_channels/test_moodle/test_client.py:39]: I know this isn't a part of your PR but maybe we can remove the unused line?

**Review by alex-sheehan-edx — APPROVED** [review]
a couple of nits but I'm G2G once those are addressed!

**Pull request merged**